src/storage/src/source/kafka.rs

```diff
@@ -817,11 +819,27 @@ where
     C: ConsumerContext,
 {
     let metadata = consumer.fetch_metadata(Some(topic), timeout)?;
-    Ok(metadata.topics()[0]
-        .partitions()
-        .iter()
-        .map(|x| x.id())
-        .collect())
+    let topic_meta = metadata
```

Comment:
How'd you find this problem? Philip's tests?

Reply:
~Yes: Philip's test covered the case when the topic was deleted, but this method would return successfully.

I haven't seen all of the new checks in this method fail in practise (and eg. I'm not sure if Kafka will ever return an empty topic list) but it seems wise to at least check!

Reply:
> but it seems wise to at least check!
💯